.github/workflows/go.yaml

```diff
@@ -0,0 +1,246 @@
+        uses: actions/checkout@v3
+      - name: Create Cache DIR
+        run: mkdir -p /home/runner/.cache/go-build
+        continue-on-error: true
+
+      - name: Checkout custom actions
+        uses: actions/checkout@v3
+        with:
+          repository: CyberOwlTeam/custom-github-actions
+          ref: release/0.0.1
+          token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }} # `GH_PAT` is a secret that contains your PAT
+          path: custom-github-actions
+
+      - name: generate branch tag step
+        uses: ./custom-github-actions/generate-branch-tag
+        id: generate-branch-tag
+      # Use the output from the `generate branch tag` step
+      - name: Get the branch tag
+        run: echo "The branch tag is  ${{ steps.generate-branch-tag.outputs.release_tag }}"
+
+  go-test:
+    name: Go Test
+    needs: [start-runner, checkout] # start-runner # required to start the main job when the runner is ready
+    runs-on: ${{ needs.start-runner.outputs.label }} # run the job on the newly created runner
+    env:
+      HOME: "/home/runner"
+      TESTCONTAINERS_RYUK_DISABLED: "true"
+      GOPATH: "/var/go"
+
+    steps:
+      - name: set PATH
+        id: set_path
+        run: echo "$GOPATH/bin:/opt/pact/bin" >> $GITHUB_PATH
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+        env:
+          GOCACHE: /home/runner/.cache/go-build
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.22.1'
+          cache: true
+          cache-dependency-path: 'go.sum'
+        env:
+          GOCACHE: /home/runner/.cache/go-build
+
+      - name: Create Docker Config
+        id: create_docker_config
+        run: |
+          mkdir -p $HOME/.docker/
+          echo '${{ vars.DOCKER_CONFIG }}' > $HOME/.docker/config.json
+
+      - name: Display Go Version
+        id: go_version
+        run: |
+          whoami
+          echo $PATH
+          ls -altr $GOPATH/*
+          go version
+        shell: bash
+
+      - name: Run Test
+        id: run_test
+        run: make all
+        shell: bash
+        env:
+          HOME: /home/runner
+          GOPATH: /var/go
+          PATH: $PATH:/usr/local/go/bin:$GOPATH/bin:/usr/sbin:/usr/bin:/sbin:/bin
+          GOCACHE: /home/runner/.cache/go-build
+        continue-on-error: true
+
+      - name: Remove Docker Config
+        id: remove_docker_config
+        run: |
+          rm -rf $HOME/.docker
+
+      - name: Slack Status Code Build and Test Success to dev-builds
+        if: steps.run_test.outcome == 'success'
+        uses: slackapi/slack-github-action@v1.24.0
+        with:
+          # Unlike the step posting a new message, this step does not accept a channel name.
+          # Please use a channel ID, not a name here.
+          channel-id: 'CJX8N7FV5'
```

Comment:
Probably worth it to use a secret here? 🤔

Reply:
all the "secret stuff" doesn't quite work the same with a public repo, so we've gone back to the original gha script, and I've locked down who can run it for the moment (to only CyberOwl users). We can open it up further if/when the need arises.